### PR TITLE
Set default icons, make icon label matching safer

### PR DESCRIPTION
**Pull request by wow-such-code**

### CODE REVIEW

**Review by KochTobi — CHANGES_REQUESTED** [review]

**Review by Steffengreiner — DISMISSED** [review]
Heya @wow-such-code nice fix however i'd recommend to also set the icon fields as required with a helper text similiar to the specimen multiselectcombobox. 
Additionally please ensure that a possible null value is caught correctly (or replace the combobox with a selection component)
(You can test this via deleting the selected default value manually, which will trigger the exception again)

**Comment by sven1103** [comment]
Addresses #618

File: user-interface/src/main/java/life/qbic/datamanager/views/projects/create/BioIconComboboxFactory.java [hunk 1]
```diff
@@ -26,8 +29,15 @@ public ComboBox<BioIcon> iconBox(List<BioIcon> options, String title) {
 
       return element;
     }));
-    comboBox.addValueChangeListener(valueChanged -> valueChanged.getSource()
-        .setPrefixComponent(styleIcon(valueChanged.getValue())));
+    comboBox.addValueChangeListener(valueChanged -> {
+      Object val = valueChanged.getValue();
+      if(val==null) {
+        comboBox.setValue(defaultIcon);
```
[the remark this answers is not in the document]
Comment (wow-such-code) [reply user-interface/src/main/java/life/qbic/datamanager/views/projects/create/BioIconComboboxFactory.java:35]: I tested it and it worked, so I would assume it does

File: user-interface/src/main/java/life/qbic/datamanager/views/projects/create/BioIconComboboxFactory.java [hunk 2]
```diff
@@ -7,15 +7,18 @@
...
 import life.qbic.datamanager.views.projects.project.experiments.experiment.ExperimentDetailsComponent.BioIcon;
+import life.qbic.datamanager.views.projects.project.experiments.experiment.ExperimentDetailsComponent.SampleSourceType;
 
 /**
  * Factory class for creating ComboBoxes allowing the selection of icons for species and specimen.
  */
 public class BioIconComboboxFactory {
 
-  public ComboBox<BioIcon> iconBox(List<BioIcon> options, String title) {
+  public ComboBox<BioIcon> iconBox(SampleSourceType type, String title) {
```
[the remark this answers is not in the document]
Comment (wow-such-code) [reply user-interface/src/main/java/life/qbic/datamanager/views/projects/create/BioIconComboboxFactory.java:17]: yeah, maybe I should add another constructor? It's hard to determine a default value without some context information, though

**Pull request merged**